app/src/test/java/org/mozilla/fenix/components/PrivateShortcutCreateManagerTest.kt

```diff
@@ -0,0 +1,92 @@
+
+@RunWith(FenixRobolectricTestRunner::class)
+class PrivateShortcutCreateManagerTest {
+
+    @Before
+    fun setup() {
+        mockkStatic(ShortcutManagerCompat::class)
+        mockkStatic(PendingIntent::class)
+    }
+
+    @Test
+    fun `GIVEN shortcut pinning is not supported WHEN createPrivateShortcut is called THEN do not create a pinned shortcut`() {
+
+        every { ShortcutManagerCompat.isRequestPinShortcutSupported(testContext) } returns false
+
+        PrivateShortcutCreateManager.createPrivateShortcut(testContext)
+
+        verify(exactly = 0) { ShortcutManagerCompat.requestPinShortcut(testContext, any(), any()) }
+    }
+
+    @Test
+    fun `GIVEN shortcut pinning is supported WHEN createPrivateShortcut is called THEN create a pinned shortcut`() {
+        val shortcut = slot<ShortcutInfoCompat>()
+        val intentSender = slot<IntentSender>()
+        val intent = slot<Intent>()
+
+        every { ShortcutManagerCompat.isRequestPinShortcutSupported(testContext) } returns true
+
+        PrivateShortcutCreateManager.createPrivateShortcut(testContext)
+
+        verify { PendingIntent.getActivity(testContext, 0, capture(intent), PendingIntent.FLAG_UPDATE_CURRENT) }
+        verify { ShortcutManagerCompat.requestPinShortcut(testContext, capture(shortcut), capture(intentSender)) }
+        `assert shortcutInfoCompat is build correctly`(shortcut.captured)
+        `assert homeScreenIntent is built correctly`(intent.captured)
+    }
+
+    private fun `assert shortcutInfoCompat is build correctly`(shortcutInfoCompat: ShortcutInfoCompat) {
+        assertEquals(testContext.getString(R.string.app_name_private_5, testContext.getString(R.string.app_name)), shortcutInfoCompat.shortLabel)
+        assertEquals(testContext.getString(R.string.app_name_private_5, testContext.getString(R.string.app_name)), shortcutInfoCompat.longLabel)
+        assertEquals(R.mipmap.ic_launcher_private_round, shortcutInfoCompat.icon.resId)
+        `assert homeActivity intent is built correctly`(shortcutInfoCompat.intent)
+    }
+
+    private fun `assert homeActivity intent is built correctly`(intent: Intent) {
+        assertEquals(Intent.ACTION_VIEW, intent.action)
+        assertEquals(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK, intent.flags)
+        assertEquals(HomeActivity::class.qualifiedName, intent.component?.className)
+        assertEquals(true, intent.extras?.getBoolean(HomeActivity.PRIVATE_BROWSING_MODE))
+        assertEquals(StartSearchIntentProcessor.PRIVATE_BROWSING_PINNED_SHORTCUT, intent.extras?.getString(HomeActivity.OPEN_TO_SEARCH))
+    }
+
+    private fun `assert homeScreenIntent is built correctly`(intent: Intent) {
+        assertEquals(Intent.ACTION_MAIN, intent.action)
+        assert(intent.categories.contains(Intent.CATEGORY_HOME))
+        assertEquals(Intent.FLAG_ACTIVITY_NEW_TASK, intent.flags)
+    }
+
+    @After
+    fun tearDown() {
```

Comment:
Can you please move this above the other public `@Test` methods?.
Seeing `@Before` and `@After` above all tests helps with readability & file navigation.

Reply:
understood. I preferred at bottom in my personal projects. Thats why added in bottom